tools/cli/src/mentat_cli/command_parser.rs

```diff
@@ -224,6 +265,112 @@ mod tests {
     }
 
     #[test]
+    fn test_query_parser_complete_edn() {
+        let input = ".q [:find ?x :where [?x foo/bar ?y]]";
+        let cmd = command(&input);
+        match cmd {
+            Command::Query(edn) => assert_eq!(edn, "[:find ?x :where [?x foo/bar ?y]]"),
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_query_parser_incomplete_edn() {
+        let input = ".q [:find ?x\r\n";
+        let cmd = command(&input);
+        match cmd {
+            Command::Query(edn) => assert_eq!(edn, "[:find ?x\r\n"),
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_query_parser_empty_edn() {
+        let input = ".q {}";
+        let cmd = command(&input);
+        match cmd {
+            Command::Query(edn) => assert_eq!(edn, "{}"),
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_query_parser_no_edn() {
+        let input = ".q ";
+        let cmd = command(&input);
+        match cmd {
+            Command::Close => assert!(true),
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_query_parser_invalid_start_char() {
+        let input = ".q :find ?x";
+        let cmd = command(&input);
+        match cmd {
+            Command::Err(message) => {
+                assert_eq!(message, format!("Invalid command {:?}", input));
+            },
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    fn test_transact_parser_complete_edn() {
+        let input = ".t [:db/add \"s\" :db/ident :foo/uuid]";
```

Comment:
A transaction is a vector-of-vectors or a vector-of-maps, in order to handle multiple assertions.  So
```rust
.t [[:db/add "t" ...]
     [:db/add "r" ...]]
```